aws/iam_group_test.go

```diff
@@ -0,0 +1,171 @@
+package aws
+
+import (
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"testing"
+	"time"
+
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/gruntwork-io/cloud-nuke/config"
+	"github.com/gruntwork-io/cloud-nuke/util"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// Test that we can list IAM groups in an AWS account
+func TestListIamGroups(t *testing.T) {
+	t.Parallel()
+
+	region, err := getRandomRegion()
+	require.NoError(t, err)
+
+	localSession, err := session.NewSession(&awsgo.Config{
+		Region: awsgo.String(region),
+	})
+	require.NoError(t, err)
+
+	groupNames, err := getAllIamGroups(localSession, time.Now(), config.Config{})
+	require.NoError(t, err)
+	assert.NotEmpty(t, groupNames)
+}
+
+// Creates an empty IAM group for testing
+func createEmptyTestGroup(t *testing.T, session *session.Session, name string) error {
+	svc := iam.New(session)
+
+	groupInput := &iam.CreateGroupInput{
+		GroupName: awsgo.String(name),
+	}
+
+	_, err := svc.CreateGroup(groupInput)
+	require.NoError(t, err)
+	return nil
+}
+
+// Stores information for cleanup
+type groupInfo struct {
+	GroupName *string
+	UserName  *string
+	PolicyArn *string
+}
+
+func createNonEmptyTestGroup(t *testing.T, session *session.Session, groupName string, userName string) (*groupInfo, error) {
+	svc := iam.New(session)
+
+	//Create User
+	userInput := &iam.CreateUserInput{
+		UserName: awsgo.String(userName),
+	}
+
+	_, err := svc.CreateUser(userInput)
+	require.NoError(t, err)
+
+	//Create Policy
+	policyOutput, err := svc.CreatePolicy(&iam.CreatePolicyInput{
+		PolicyDocument: awsgo.String(`{
+			"Version": "2012-10-17",
+			"Statement": [
+					{
+							"Sid": "VisualEditor0",
```

Comment:
NIT: `Sid` is another place you could feel free to put some more debug / Grunt-facing info about this policy (as you did in the description), if you wanted: https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_sid.html. 

Might make it easier to visually identity what we're looking at once we're digging through an AWS account debugging.